backend/database_handler/transactions_processor.py

```diff
@@ -100,22 +102,20 @@ def insert_transaction(
             r=None,
             s=None,
             v=None,
+            leader_only=leader_only,
         )
 
         self.session.add(new_transaction)
 
-        self.session.flush()  # SQLAlchemy will populate all the fields that are set by the database, like the id and created_at fields
+        self.session.flush()  # So that `created_at` gets set
 
-        # Insert transaction audit record into the transactions_audit table
         transaction_audit_record = TransactionsAudit(
             transaction_hash=new_transaction.hash,
             data=self._parse_transaction_data(new_transaction),
         )
 
         self.session.add(transaction_audit_record)
 
-        self.session.commit()
-
         return new_transaction.hash
```

Comment:
I'm guessing it was a mistake to commit here in the first place, hence the removal?

Reply:
exactly! I've mentioned it to Cristiam. It's not "wrong" to `commit` every time, but in general it's better to leave the session committing/rollback to flask so that it does it per request. We have that done with the new code here 

https://github.com/yeagerai/genlayer-simulator/blob/273c0d8607f6e9174d2505d2aa50763cf826ce4f/backend/protocol_rpc/server.py#L96-L105